### PR TITLE
[5.6] Filesystem changes

### PR DESCRIPTION
**Pull request by Smolevich**
Refactoring methods get, getRequire, put, prepend, append, delete. Methods can write or read files, add check permissions to read or write.
For example, 
```
use Illuminate\Filesystem\Filesystem;

$fileSystem = new Filesystem();
$res = $fileSystem->put('/test/test.txt', 'data'); //directory /test has permissions 0400 for user root, but script executed by not root-user 
var_dump($res); // output: bool(false)
``` 
In version before changes script failed with fatal error ` Uncaught ErrorException: file_put_contents(/test/test.txt): failed to open stream: No such file or directory`

### CODE REVIEW

File: src/Illuminate/Filesystem/Filesystem.php [hunk 1]
```diff
@@ -115,19 +115,25 @@ public function hash($path)
      * @param  string  $path
      * @param  string  $contents
      * @param  bool  $lock
-     * @return int
+     * @return int|bool
      */
     public function put($path, $contents, $lock = false)
     {
-        return file_put_contents($path, $contents, $lock ? LOCK_EX : 0);
+        $result = false;
```
Comment (TBlindaruk) [inline src/Illuminate/Filesystem/Filesystem.php]: let`s $result = null

Comment (Smolevich) [reply src/Illuminate/Filesystem/Filesystem.php]: Why null?

File: src/Illuminate/Filesystem/Filesystem.php [hunk 2]
```diff
@@ -115,19 +115,27 @@ public function hash($path)
      * @param  string  $path
      * @param  string  $contents
      * @param  bool  $lock
-     * @return int
+     * @return int|bool
```
Comment (GrahamCampbell) [inline src/Illuminate/Filesystem/Filesystem.php]: Why does this have to change?

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Illuminate/Filesystem/Filesystem.php (hunk 2 above)
Comment (36864) [reply src/Illuminate/Filesystem/Filesystem.php]: From the [php docs](http://php.net/manual/en/function.file-put-contents.php) for file_put_contents:

>This function returns the number of bytes that were written to the file, or FALSE on failure.

**Comment by taylorotwell** [comment]
I am closing this pull request because it lacks sufficient explanation, tests, or both. It is difficult for us to merge pull requests without these things because the change may introduce breaking changes to the framework.

Feel free to re-submit your change with a **thorough** explanation of the feature and tests - integration tests are preferred over unit tests. Please include it's benefit to end users; the reasons it does not break any existing features; how it makes building web applications easier, etc.

Thanks!

**Comment by Smolevich** [comment]
What integration tests do you mean? I think if function has php_docs with one exception in section throw, then throw not catched exception `ErrorException` it is not correct

**Pull request closed without merging**